component/azstorage/block_blob.go

```diff
@@ -590,32 +588,39 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 		return blobList, nil, err
 	}
 
-	dereferenceTime := func(input *time.Time, defaultTime time.Time) time.Time {
-		if input == nil {
-			return defaultTime
-		} else {
-			return *input
-		}
-	}
-
 	// Process the blobs returned in this result segment (if the segment is empty, the loop body won't execute)
 	// Since block blob does not support acls, we set mode to 0 and FlagModeDefault to true so the fuse layer can return the default permission.
 
 	// For some directories 0 byte meta file may not exists so just create a map to figure out such directories
 	var dirList = make(map[string]bool)
 
 	for _, blobInfo := range listBlob.Segment.BlobItems {
+		if blobInfo.Name == nil {
+			continue
+		}
+
 		attr := &internal.ObjAttr{
-			Path:   split(bb.Config.prefixPath, blobInfo.Name),
-			Name:   filepath.Base(blobInfo.Name),
-			Size:   *blobInfo.Properties.ContentLength,
-			Mode:   0,
-			Mtime:  blobInfo.Properties.LastModified,
-			Atime:  dereferenceTime(blobInfo.Properties.LastAccessedOn, blobInfo.Properties.LastModified),
-			Ctime:  blobInfo.Properties.LastModified,
-			Crtime: dereferenceTime(blobInfo.Properties.CreationTime, blobInfo.Properties.LastModified),
-			Flags:  internal.NewFileBitMap(),
-			MD5:    blobInfo.Properties.ContentMD5,
+			Path:  split(bb.Config.prefixPath, *blobInfo.Name),
+			Name:  filepath.Base(*blobInfo.Name),
+			Mode:  0,
+			Flags: internal.NewFileBitMap(),
+			MD5:   blobInfo.Properties.ContentMD5,
+		}
+
+		if blobInfo.Properties.ContentLength != nil {
```

Comment:
These if conditions will slow down the system when we are listing large number of objects as they will be hit for each item. If we do not hit a situation where these are nil, may be we can ignore these checks.

Reply:
Removed the nil checks